src/app/shared/pagination/pagination.directive.ts

```diff
@@ -0,0 +1,93 @@
+import {Directive, Input, OnInit, TemplateRef, ViewContainerRef} from '@angular/core';
+import {BehaviorSubject, Subject, filter, map, takeUntil} from 'rxjs';
+import {PaginationContext} from './pagination-context.interface';
+import {createPageIndexes, sliceGroup} from './utils';
+
+@Directive({
+    selector: '[appPagination]',
+})
+export class PaginationDirective<T> implements OnInit {
+    @Input() appPaginationOf: T[] | undefined | null;
+    @Input() appPaginationChankSize = 0;
+    private readonly currentIndex$ = new BehaviorSubject<number>(0);
+    private readonly destroy$ = new Subject<void>();
+    private pageIndexes: number[] | null = null;
+
+    constructor(
+        private readonly viewContainerRef: ViewContainerRef,
+        private readonly templateRef: TemplateRef<PaginationContext<T>>,
+    ) {}
+
+    private setPageIndexes(): void {
+        if (!this.appPaginationOf || !this.appPaginationOf.length) {
```

Comment:
На мой взгляд, можно было бы сделать чуть проще и не менее прозрачнее
```suggestion
        if (!this.appPaginationOf?.length) {
```

Reply:
Согласен,  сделал немного по другому